src/hooks/useCountdownTimer.tsx

```diff
@@ -0,0 +1,109 @@
+import { useContext, useEffect, useRef, useState } from 'react';
+import { TimersContext } from '../views/TimerProvider';
+
+export const useCountdownTimer = (totalSeconds: number, isActive: boolean, onFinish: () => void, totalReps: number, totalSecondsRest: number) => {
+    const { resetAll, setTotalSecondsPassed } = useContext(TimersContext);
+
+    const [secondsPassed, setSecondsPassed] = useState<number>(0);
+    const [repsRemaining, setRepsRemaining] = useState<number>(totalReps);
+
+    const oneRoundSecondsRest = totalSecondsRest / totalReps;
+    const oneRoundSecondsWork = totalSeconds / totalReps - oneRoundSecondsRest;
+    const oneRoundSeconds = oneRoundSecondsRest + oneRoundSecondsWork;
+    const [oneRoundSecondsLeft, setOneRoundSecondsLeft] = useState<number>(oneRoundSeconds);
+    const [isWorkPhase, setIsWorkPhase] = useState(true);
+
+    const fastforward = () => {
+        if (intervalRef.current !== null) {
+            clearInterval(intervalRef.current);
+            intervalRef.current = null;
+        }
+        const remainingTime = totalSeconds - secondsPassed;
+        setSecondsPassed(totalSeconds);
+        setRepsRemaining(0);
+        setTotalSecondsPassed((prevTotal: number) => prevTotal + remainingTime);
+        onFinish();
+    };
+
+    const intervalRef: React.MutableRefObject<number | null> = useRef(null);
+
+    useEffect(() => {
+        if (resetAll) {
+            setSecondsPassed(0);
+            setOneRoundSecondsLeft(oneRoundSeconds);
+            setRepsRemaining(totalReps);
+        }
+    }, [resetAll, totalReps, oneRoundSeconds]);
+
+    useEffect(() => {
+        if (isActive) {
+            if (intervalRef.current !== null) {
```

Comment:
I believe you're recreating the interval every second. If this is your implementation use `setTimeout` instead, which runs once instead of infinite. Its safer in case there's an edge condition that skips clearing it.